makefile

```diff
@@ -25,7 +25,7 @@ C++11 = false
 ##
 # Set default compiler options.
 ## 
-CFLAGS = -I src -isystem $(EIGEN) -isystem $(BOOST) -isystem $(CVODES)/include -isystem $(MATH) -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -DNO_FPRINTF_OUTPUT -pipe 
+CFLAGS = -I src -isystem $(EIGEN) -isystem $(BOOST) -isystem $(CVODES)/include -isystem $(MATH) -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DFUSION_MAX_VECTOR_SIZE=12 -DNO_FPRINTF_OUTPUT -pipe -Wno-unused-local-typedefs
```

Comment:
I'm worried about different versions of gcc, not necessarily clang. If I remember correctly, bad options will make it kick out.

Re: `make/cc_gcc`, I think I'm not really a fan. I'm looking at the options and there's a serious interaction between OS and compiler. If we go in that direction, what makes sense is to do: `make/win_gcc`, `make/win_clang`, etc. I'd be more inclined to (as much as it hurts saying this) repeat the compiler option in 3 places (`os_*`) and be done with it. We don't change it that often.